msticpy/data/drivers/sentinel_query_reader.py

```diff
@@ -0,0 +1,452 @@
+            if file.startswith(  # type: ignore
+                (
+                    "Azure-Sentinel-master/Detections/",
+                    "Azure-Sentinel-master/Hunting Queries/",
+                )
+            ):
+                archive.extract(file, path=outputdir)  # type: ignore
+        print("Downloaded and Extracted Files successfully")
+        return True
+
+    except httpx.HTTPError as http_err:
+        warnings.warn(f"HTTP error occurred trying to download from Github: {http_err}")
+        return False
+
+
+def read_yaml_files(parent_dir: str, child_dir: str) -> dict:
+    """
+    Create dictionary mapping query file paths with the yaml file text each contains.
+
+    Parameters
+    ----------
+    parent_dir : str
+        Directory storing the Hunting and Detections directories
+    child_dir : str
+        Either "Hunting Queries" or "Detections" or otherwise named query category
+
+    Returns
+    -------
+    dict
+       Dictionary mapping query file paths to corresponding yaml file text in the
+       parent_dir/child_dir specified. Only identifies .yaml files.
+
+    """
+    # enumerate the files and read the yaml
+    yaml_queries = glob.glob(f"{parent_dir}/{child_dir}/**/*.yaml", recursive=True)
+    yaml_queries = [str(Path(q)) for q in yaml_queries]
+    parsed_query_dict = {}
+
+    for query in yaml_queries:
+        with open(query, encoding="utf8", errors="ignore") as opened_query_file:
+            parsed_query_dict[query] = opened_query_file.read()
+
+    return parsed_query_dict
+
+
+def import_sentinel_queries(yaml_files: dict, query_type: str) -> list:
+    """
+    Create list of SentinelQuery attr objects.
+
+    Parameters
+    ----------
+    yaml_files : dict
+        Dictionary mapping query file addresses to yaml file text created by read_yaml_files
+    query_type : str
+        Either "Hunting Queries" or "Detections" or otherwise named query category
+
+    Returns
+    -------
+    list
+        Returns a list of SentinelQuery attr objects from a dict of yaml files and query type given
+
+    """
+    return [
+        _import_sentinel_query(yaml_path, yaml_text, query_type)
+        for yaml_path, yaml_text in yaml_files.items()
+    ]
+
+
+def _import_sentinel_query(
+    yaml_path: str, yaml_text: str, query_type: str
+) -> SentinelQuery:
+    """
+    Create a SentinelQuery attr object for a given yaml query.
+
+    Parameters
+    ----------
+    yaml_path : str
+        File path to a YAML Sentinel query
+    yaml_text : str
+        YAML text (Sentinel query) associated with the yaml_path
+    query_type : str
+        Either "Hunting Queries" or "Detections" or otherwise named query category
+
+    Returns
+    -------
+    SentinelQuery
+        Returns an attrs object called SentinelQuery with all the YAML query information
+
+    """
+    try:
+        parsed_yaml_dict = yaml.load(yaml_text, Loader=yaml.SafeLoader)
+        new_query = SentinelQuery(
```

Comment:
Just a thought here.
You could implement the SentinelQuery using exactly the same names as the source data (most already are)
Set defaults on the ones that need defaults ("",, Factory(list), Factory(dict), etc.
Then you can just call it like this
```
    new_query = SentinelQuery(**parsed_yaml_dict)
```

You can add additional params as keywords
```
    new_query = SentinelQuery(**parsed_yaml_dict, source_file=yaml_path, query_type=query_type)
```

Reply:
I didn't use the exact same names because the camel case didn't match what the Python linters were looking for. Totally happy to change this back to the original source name but keeping it for now - let me know if it would be better to switch over!